osdep/io.c

```diff
@@ -902,3 +902,32 @@ int mp_mkostemps(char *template, int suffixlen, int flags)
     errno = EEXIST;
     return -1;
 }
+
+bool mp_save_to_file(const char *filepath, const void *data, size_t size)
+{
+    assert(filepath && data && size);
```

Comment:
this function assumes that `size > 0` but shouldn't it be valid to write an empty file?

Reply:
If you want to clear file just fopen/fclose it, there is no reason to create temp file move it and so on. This functions is meant to write data to file.